src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -662,31 +728,83 @@ Json::Value STAmount::getJson(JsonOptions) const
 void
 STAmount::add(Serializer& s) const
 {
-    if (mIsNative)
-    {
-        assert(mOffset == 0);
-
-        if (!mIsNegative)
-            s.add64(mValue | cPosNative);
-        else
-            s.add64(mValue);
-    }
-    else
+    switch (mType)
     {
-        if (*this == beast::zero)
-            s.add64(cNotNative);
-        else if (mIsNegative)  // 512 = not native
-            s.add64(
-                mValue |
-                (static_cast<std::uint64_t>(mOffset + 512 + 97) << (64 - 10)));
-        else  // 256 = positive
-            s.add64(
-                mValue |
-                (static_cast<std::uint64_t>(mOffset + 512 + 256 + 97)
-                 << (64 - 10)));
-
-        s.addBitString(mIssue.currency);
-        s.addBitString(mIssue.account);
+        case STAmount::Type::xrp:
+            /*
+             * Serialized format:
+             *
+             * Value encoding (first 64 bits):
+             *  0 [sign bit] 0 [4 unused bits, set to 0] [57 bits for value]
+             */
+            // TODO why?
+            assert(mOffset == 0);
+            // Ensure that value is no greater than 57 bits
+            assert(mValue == (mValue & cValueMask));
```

Comment:
question

Reply:
the comment was intended to say what the purpose is here. this asserts that `mValue` is not greater than 57 bits.